pyvista/plotting/charts.py

```diff
@@ -0,0 +1,3375 @@
+        scatter_plot, line_plot = None, None
+        if marker_style != "":
+            scatter_plot = self.scatter(x, y, color, style=marker_style)
+        if line_style != "":
+            line_plot = self.line(x, y, color, style=line_style)
+        return scatter_plot, line_plot
+
+    def scatter(self, x, y, color="b", size=10, style="o", label=""):
+        """Add a scatter plot to this chart.
+
+        See ``pyvista.ScatterPlot2D`` for an overview of the possible parameters.
+
+        Examples
+        --------
+        Generate a scatter plot.
+
+        >>> import pyvista
+        >>> chart = pyvista.Chart2D()
+        >>> plot = chart.scatter([0, 1, 2], [2, 1, 3])
+        >>> chart.show()
+
+        """
+        return self._add_plot("scatter", x, y, color=color, size=size, style=style, label=label)
+
+    def line(self, x, y, color="b", width=1.0, style="-", label=""):
+        """Add a line plot to this chart.
+
+        See ``pyvista.LinePlot2D`` for an overview of the possible parameters.
+
+        Examples
+        --------
+        Generate a line plot.
+
+        >>> import pyvista
+        >>> chart = pyvista.Chart2D()
+        >>> plot = chart.line([0, 1, 2], [2, 1, 3])
+        >>> chart.show()
+
+        """
+        return self._add_plot("line", x, y, color=color, width=width, style=style, label=label)
+
+    def area(self, x, y1, y2=None, color="b", label=""):
+        """Add an area plot to this chart.
+
+        See ``pyvista.AreaPlot`` for an overview of the possible parameters.
+
+        Examples
+        --------
+        Generate an area plot.
+
+        >>> import pyvista
+        >>> chart = pyvista.Chart2D()
+        >>> plot = chart.area([0, 1, 2], [2, 1, 3])
+        >>> chart.show()
+
+        """
+        return self._add_plot("area", x, y1, y2, color=color, label=label)
+
+    def bar(self, x, y, color=None, offset=5, orientation="V", label=None):
+        """Add a bar plot to this chart.
+
+        See ``pyvista.BarPlot`` for an overview of the possible parameters.
+
+        Examples
+        --------
+        Generate a bar plot.
+
+        >>> import pyvista
+        >>> chart = pyvista.Chart2D()
+        >>> plot = chart.bar([0, 1, 2], [2, 1, 3])
+        >>> chart.show()
+
+        """
+        return self._add_plot("bar", x, y, color=color, offset=offset, orientation=orientation, label=label)
+
+    def stack(self, x, ys, colors=None, labels=None):
+        """Add a stack plot to this chart.
+
+        See ``pyvista.StackPlot`` for an overview of the possible parameters.
+
+        Examples
+        --------
+        Generate a stack plot.
+
+        >>> import pyvista
+        >>> chart = pyvista.Chart2D()
+        >>> plot = chart.stack([0, 1, 2], [[2, 1, 3],[1, 2, 1]])
+        >>> chart.show()
+
+        """
+        return self._add_plot("stack", x, ys, colors=colors, labels=labels)
+
+    def plots(self, plot_type=None):
+        """Return all plots (of the specified type) in this chart.
```

Comment:
I assume the linter complains about "Return...", thinking that you're specifying inputs and outputs on this line. Arguably a false positive. Perhaps you could use "Retrieve" here as well to make it happy.

But `plot_type` is undocumented, consider adding that. Although I can see that all the others are missing too, from a glance, so that might just be a WIP.

Reply:
All right thanks for the feedback! I'm going to focus on the tests and remaining todos now, then I'll do a final pass through all the changes to fix the last missing parts :)

Reply:
Hmmm, "Retrieve" didn't do the trick, same error message :/

Reply:
Ok, so apparently it's the parenthesis that are causing this. Although I quite liked them because the parameter is optional.

Reply:
Ehh, that's dumb of the linter then. There's probably a way to make an exception for that docstring, but I guess the line makes sense even without parentheses. Just explain optionalness later in the docstring.